tools/ir-generator/irclass.cpp

```diff
@@ -222,15 +223,30 @@ void IrDefinitions::generate(std::ostream &t, std::ostream &out, std::ostream &i
 
     // Add some specials:
     t << "  IDeclaration = " << nkId++ << ",\n";
-    t << "  VectorBase = " << nkId++ << ",\n";
-    t << "  VectorT = UINT64_C(1) << 56,\n"
-      << "  IndexedVectorT = UINT64_C(2) << 56,\n"
-      << "  FirstAuto = UINT64_C(0xFF) << 56\n"
+    t << "  VectorBase = " << nkId++ << "\n"
+      << "};\n";
+    t << "enum class NodeDiscriminator : RTTI::TypeId {\n"
+      << "  NodeT = UINT64_C(1),\n"
+      << "  VectorT = UINT64_C(1),\n"
+      << "  IndexedVectorT = UINT64_C(2),\n"
+      << "  FirstAuto = UINT64_C(0xFF)\n"
```

Comment:
Maybe just `Auto` as this is no longer first, all the auto-values have this discriminator.

Reply:
Done